src/base/utils/string.cpp

```diff
@@ -130,6 +130,20 @@ bool NaturalCompare::lessThan(const QString &left, const QString &right)
     return false;
 }
 
+bool Utils::String::naturalCompareCaseSensitive(const QString &left, const QString &right)
+{
+    // provide a single `NaturalCompare` instance for easy use
+    static NaturalCompare nCmp(true);  // this is thread-safe in C++11 (stated in spec 6.7.4)
+    return nCmp(left, right);
+}
+
+bool Utils::String::naturalCompareCaseInsensitive(const QString &left, const QString &right)
+{
+    // provide a single `NaturalCompare` instance for easy use
+    static NaturalCompare nCmp(false);  // this is thread-safe in C++11 (stated in spec 6.7.4)
```

Comment:
>Since QCollator::compare is const it does not cause copying.

You made another point, if it's declared with `const` specifier, then no class member (member of `QCollator`) can be alternated when invoking that function. Then it should imply the function is thread-safe.